core/commands/dns.go

```diff
@@ -63,7 +63,7 @@ The resolver will give:
 			depth = namesys.DefaultDepthLimit
 		}
 		output, err := resolver.ResolveN(req.Context().Context, name, depth)
-		if err != nil {
+		if err != nil && (err != namesys.ErrResolveRecursion || depth > 1) {
```

Comment:
there's another way to look at it:

- `resolve X` means "dereference X"
- `resolve --depth=3 X` means "dereference X 3 times"
- `resolve --recursive X` means "dereference X until you cannot dereference any more"

If the command is satisfied, that's not really an error. We may also have IPNS values that dont even point to `/ipfs/...`. I think it's up to the user to figure out what to do with the returned value. This is a simpler interface with less complexity (having to figure out when to check the error, or not, etc).

In any case, i think the "return value and an error value" will be a bit ... error ... prone. the `io.EOF` thing is the only example that comes up in my mind from the go stdlib.